File: stkpush/src/main/java/com/freddygenicho/mpesa/stkpush/api/response/STKPushResponse.java
package com.freddygenicho.mpesa.stkpush.api.response;

import com.google.gson.Gson;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * @author Fredrick Ochieng on 02/02/2018.
 */

public class STKPushResponse {

    @SerializedName("MerchantRequestID")
    @Expose
    private String merchantRequestID;
    @SerializedName("CheckoutRequestID")
    @Expose
    private String checkoutRequestID;
    @SerializedName("ResultCode")
    @Expose
    private Integer resultCode;
    @SerializedName("ResultDesc")
    @Expose
    private String resultDesc;
    @SerializedName("CallbackMetadata")
    @Expose
    private CallbackMetadata callbackMetadata;

    public String getMerchantRequestID() {
        return merchantRequestID;
    }

    public void setMerchantRequestID(String merchantRequestID) {
        this.merchantRequestID = merchantRequestID;
    }

    public String getCheckoutRequestID() {
        return checkoutRequestID;
    }

    public void setCheckoutRequestID(String checkoutRequestID) {
        this.checkoutRequestID = checkoutRequestID;
    }

    public Integer getResultCode() {
        return resultCode;
    }

    public void setResultCode(Integer resultCode) {
        this.resultCode = resultCode;
    }

    public String getResultDesc() {
        return resultDesc;
    }

    public void setResultDesc(String resultDesc) {
        this.resultDesc = resultDesc;
    }

    public CallbackMetadata getCallbackMetadata() {
        return callbackMetadata;
    }

    public void setCallbackMetadata(CallbackMetadata callbackMetadata) {
        this.callbackMetadata = callbackMetadata;
    }

    public String toString() {
        return "{\"MerchantRequestID\":\"" + merchantRequestID + "\"," +
                "\"checkoutRequestID\":\"" + checkoutRequestID + "\"," +
                "\"resultCode\":\"" + resultCode + "\"," +
                "\"resultDesc\":\"" + resultDesc + "\"," +
                "\"callbackMetadata\":\"" + callbackMetadata +
                "\"}";
    }

    public String toJson(STKPushResponse stkPushResponse) {
        Gson gson = new Gson();
        return gson.toJson(stkPushResponse);
    }

}

File: stkpush/src/main/java/com/freddygenicho/mpesa/stkpush/api/response/STKPushResponse.kt
package com.freddygenicho.mpesa.stkpush.api.response

import com.google.gson.annotations.SerializedName
import com.google.gson.annotations.Expose
import com.freddygenicho.mpesa.stkpush.api.response.CallbackMetadata
import com.freddygenicho.mpesa.stkpush.api.response.STKPushResponse
import com.google.gson.Gson

/**
 * @author Fredrick Ochieng on 02/02/2018.
 */
class STKPushResponse {
    @SerializedName("MerchantRequestID")
    @Expose
    var merchantRequestID: String? = null

    @SerializedName("CheckoutRequestID")
    @Expose
    var checkoutRequestID: String? = null

    @SerializedName("ResultCode")
    @Expose
    var resultCode: Int? = null

    @SerializedName("ResultDesc")
    @Expose
    var resultDesc: String? = null

    @SerializedName("CallbackMetadata")
    @Expose
    var callbackMetadata: CallbackMetadata? = null
    override fun toString(): String {
        return "{\"MerchantRequestID\":\"" + merchantRequestID + "\"," +
                "\"checkoutRequestID\":\"" + checkoutRequestID + "\"," +
                "\"resultCode\":\"" + resultCode + "\"," +
                "\"resultDesc\":\"" + resultDesc + "\"," +
                "\"callbackMetadata\":\"" + callbackMetadata +
                "\"}"
    }

    fun toJson(stkPushResponse: STKPushResponse?): String {
        val gson = Gson()
        return gson.toJson(stkPushResponse)
    }
}